packages/react/src/hooks/useUiElement.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from "react";
+
+export function useUiElement<State>(
```

Comment:
maybe sth like `useProsemirrorPluginEventState`?

Reply:
Hmm I think it's not gonna be super obvious what it does if we look at it in a few months time, since we use other prosemirror plugins which aren't relevant for this hook. How about just `useUiElementState`?